packages/framework/src/errors/resource.errors.ts

```diff
@@ -14,14 +14,9 @@ export abstract class ResourceNotFoundError extends NotFoundError {
   }
 }
 
-export abstract class ResourceExecutionFailed extends InternalServerError {
-  constructor(resource: ResourceEnum, id: string) {
-    super(`Failed to execute ${toPascalCase(resource)} with id: \`${id}\`. Please try again later.`);
-  }
-}
-
-export abstract class ResourcePreviewFailed extends InternalServerError {
-  constructor(resource: ResourceEnum, id: string) {
-    super(`Failed to preview ${toPascalCase(resource)} with id: \`${id}\`. Please try again later.`);
+export abstract class ResourceExecutionFailed extends ServerError {
+  statusCode = HttpStatusEnum.BAD_GATEWAY;
```

Comment:
Using a `502` error code here as I think it's the best fit - we know an error occurred during execution, and the Framework was acting as a "gateway" to the logic built by the implementor. By throwing a non-`500` error code, all the Novu Platform exception filters won't treat it as unhandled (we only treat [`500` errors as exceptions](https://github.com/novuhq/novu/blob/next/apps/api/src/exception-filter.ts#L28))